### PR TITLE
#711 AWS CLI integration

### PR DESCRIPTION
**Pull request by CREITZ25**
#711 AWS CLI integration

Documentation of aws cli

### CODE REVIEW

File: scripts/src/main/resources/scripts/command/aws [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  local path_to_package="${1}"
+  local enable_ide_update="${2}"
+  local version_file="${path_to_package}/.devon.software.version"
+  local current_version
+  current_version="$(cat "${version_file}")"
+
+  if doIsWindows
+  then
+    doEcho "Windows Installation..."
+    windows_path_to_package=$(cygpath -w "${path_to_package}")
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:36]: This is not working outside cygwin (e.g. in git-bash what is our main shell on windows).
Please talk with the team - I left analog review comments in other PRs. We should add a function that converting the current path (or a given path as $1) to the native OS path depending on OS (mainly only special for Windows) and current shell (differs for cygwin, powershell, and MinGW/git-bash).

File: scripts/src/main/resources/scripts/command/aws [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+{
+  local path_to_package="${1}"
+  local enable_ide_update="${2}"
+  local version_file="${path_to_package}/.devon.software.version"
+  local current_version
+  current_version="$(cat "${version_file}")"
+
+  if doIsWindows
+  then
+    doEcho "Windows Installation..."
```
[the remark this answers is not in the document]
Comment (hohwille) [reply scripts/src/main/resources/scripts/command/aws:35]: shouldn't `aws` be `AWS` in log messages?

File: scripts/src/main/resources/scripts/command/aws [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  then
+    doEcho "logon again to test aws with \"aws version\""
+  fi 
+
+}
+
+# CLI
+case ${1} in 
+"help" | "-h")
+  echo "Install aws."
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:100]: Should better be `"Setup or run AWS CLI."`.

File: scripts/src/main/resources/scripts/command/aws [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  echo
+  echo "Arguments:"
+  echo " setup                    install aws on your machine."
+  echo " <<args>>                 call aws with the specified arguments. Call aws --help for details or use helm directly as preferred."
+  echo
+;;
+"setup" | "s" | "")
+  doEcho "Installating aws..."
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:108]: The doSetup command should already do the logging and honor the `silent` option, so simply remove this line.

File: scripts/src/main/resources/scripts/command/aws [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  doSetup
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:109]: arguments shall be passed as in other commandlets:
https://github.com/devonfw/ide/blob/master/scripts/src/main/resources/scripts/command/mvn#L369

File: documentation/aws.asciidoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+= aws
+
+The aws commandlet allows to install and use https://aws.amazon.com/cli/[aws cli].
+
+*ATTENTION:*
+Currently this feature is new and therefore experimental.
+It may change in incompatible ways in the next releases until we reach a stable state.
+We hope that all is working fine for you.
+However, do not expect everything to work out of the box.
+In case you are facing issues (e.g. network problems with Cisco AnyConnect, etc.) please give us feedback so we can improve.
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply documentation/aws.asciidoc:13]: I deleted this part

File: documentation/aws.asciidoc [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+The arguments (`devon helm «args»`) are explained by the following table:
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply documentation/aws.asciidoc:15]: Yes, of course

File: scripts/src/main/resources/scripts/command/aws [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  current_version="$(cat "${version_file}")"
+
+  if doIsWindows
+  then
+    doEcho "Windows Installation..."
+    windows_path_to_package=$(cygpath -w "${path_to_package}")
+    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${windows_path_to_package}\aws-${current_version}-AWSCLIV2.msi /quiet'"
+  elif doIsMacOs
+  then
+    doEcho "Installating aws for MacOS..."
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:39]: At this point the code for MacOS must be added - but I don't know which commands MacOS needs. I hoped Tobias or you could support at this point

File: scripts/src/main/resources/scripts/command/aws [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    doRunCommand "cp ${version_file} ${DEVON_IDE_HOME}/software/aws/"
+  fi
+  #doRunCommand "rm -rf ${path_to_package}"
+}
+
+function doSetup() {
+
+  # start installation after extract
+  local inst_pack="y"
+  local enable_ide_update="n"
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:59]: I adjusted it to our standard

File: scripts/src/main/resources/scripts/command/aws (hunk 1 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:35]: as discussed, it works like this thanks to a recent Git Bash tweak

File: scripts/src/main/resources/scripts/command/aws (hunk 8 above)
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:39]: I added doFail command

File: scripts/src/main/resources/scripts/command/aws [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+fi
+
+if [ -n "${DEVON_IDE_TRACE}" ]; then set -vx; fi
+# shellcheck source=scripts/functions
+source "$(dirname "${0}")"/../functions
+
+# Call aws with specified arguments.
+function doRun() {
+  doSetup silent
+  doRunCommand "${AWS_HOME}/aws" "${@}"
```
[the remark this answers is not in the document]
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:20]: HOME Directory is now set as expected

File: scripts/src/main/resources/scripts/command/aws (hunk 4 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:108]: I corrected it

File: scripts/src/main/resources/scripts/command/aws (hunk 5 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:109]: Ok

**Comment by hohwille** [comment]
@CREITZ25 the build is failing because of `MigrationsTest.testMigrations:53`
If you go to the "Files Changed" tab of this PR, you can see that you have accidentally modified the git submodule at `settings/src/main/settings`. To fix the build you have to revert this change most probably by calling:
```
git submodule update
```

File: scripts/src/main/resources/scripts/command/aws (hunk 1 above)
Comment (hohwille) [reply scripts/src/main/resources/scripts/command/aws:31]: Indeed. I was wrong. Great that this also works in git-bash. Thanks for clarification!

**Pull request merged**